src/node_v8_platform-inl.h

```diff
@@ -129,8 +129,9 @@ struct V8Platform {
     // Attach a new NodeTraceWriter only if this function hasn't been called
     // before.
     if (tracing_file_writer_.IsDefaultHandle()) {
-      std::vector<std::string> categories =
-          SplitString(per_process::cli_options->trace_event_categories, ',');
+      auto out = per_process::cli_options->trace_event_categories;
+      std::vector<std::string_view> categories =
+          SplitString(std::string_view(out.data(), out.size()), ",");
 
       tracing_file_writer_ = tracing_agent_->AddClient(
           std::set<std::string>(std::make_move_iterator(categories.begin()),
```

Comment:
The `std::make_move_iterator` is possibly problematic here. There should be no need to move std::string_view instances since they are lightweight.

But also, a std::string_view does not always convert to a std::string_view. So this might give you some trouble.

Reply:
To be more precise, you can create a std::string from a string_view, but it needs to be an explicit constructor.

You could do it as a lambda:

```
 auto convert_to_set = [](std::vector<std::string_view> categories) -> std::set<std::string> {
  std::set<std::string> se;    
  for(std::string_view s : categories) {   se.emplace(s);    }
  return se;
}
```

And then...

```
tracing_file_writer_ = tracing_agent_->AddClient(convert_to_set(categories)....
```

It is a bit more code,

Reply:
Thanks. That makes sense!